src/libmongoc/tests/unified/operation.c

```diff
@@ -764,7 +847,7 @@ operation_distinct (test_t *test,
                         "{",
                         &filter,
                         "}");
-   bson_concat (distinct, bson_parser_get_extra (parser));
+   bson_concat (distinct, opts);
```

Comment:
I don't think this works. `mongoc_client_session_append` is going to append a "sessionId" to `opts`, but that's only the internal libmongoc identifier and should not end up in the raw command document being sent to the server.

This test currently uses [`mongoc_collection_command_simple`](http://mongoc.org/libmongoc/current/mongoc_collection_command_simple.html), which doesn't support sessions. I think you'll need to change it to use [`mongoc_database_read_command_with_opts`](http://mongoc.org/libmongoc/current/mongoc_database_read_command_with_opts.html) instead. And in doing so, you should pass `opts` as its own parameter (removing `bson_concat` entirely).

FWIW, `operation_distinct` appears to be the only operation to use `mongoc_collection_command_simple` so I don't think this is a concern anywhere else.

Reply:
Gotcha, thanks for the guidance. I believe you meant `mongoc_collection_read_command_with_opts`, so I switched to that with a separate `opts` parameter and no `bson_concat`.

Reply:
I was definitely referring to `mongoc_database_read_command_with_opts`. I forgot there were similarly named collection methods, but if you look at both functions you'll see they're just syntactic sugar for `_mongoc_client_command_with_opts`.

In my mind, commands are scoped to a database, not a collection, so I don't think about having a helper at the collection level. The command itself _may_ refer to a collection name (e.g. `find: <name>`), but that's still something you'd have to manually specify in the command document.

Anywho, either method should work. And since `distinct` is a collection-level operation, it makes sense that you'll already have the necessary collection object on hand here.

Reply:
Ah gotcha! I'll just use the `mongoc_collection_read_command_with_opts` since I have the collection handle already.

Reply:
Good point. As it is, `operation_distinct` does not really test any C driver functionality (and I don't think it's documented that users should create their own distinct [in this manner](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/tests/unified/operation.c#L759-L766)).

The Go and CXX test runners also error on unsupported operations. I think it would be ok to skip `distinct` tests since the C driver has no helper, but always skipping on unsupported operations might make it harder to recognize when a new operation requires a new case in the test runner.

Reply:
I'd be OK with leaving this as-is and then opening a separate CDRIVER issue to follow-up. That issue can propose an explicit skip for tests utilizing `distinct` operations. Alternatively, you might decide to utilize the `is_test_skipped` function I'm adding in https://github.com/mongodb/mongo-c-driver/pull/801 and just explicitly add tests using `distinct` to that list.

Either way, I think it makes sense to figure out how we want to handle this _before_ the retryable read tests get ported and we need to skip its tests for unsupported operations (e.g. collection enumeration variants). In that case, the explicit list in `is_test_skipped` probably makes the most sense.

Reply:
Filed CDRIVER-4024. Using something like `is_test_skipped` seems best to me; then we can explicitly list the tests with unsupported operations like `distinct` or collection enumeration variants. Leaving `operation_distinct` as-is for now, though.